packages/gestalt/src/PageHeaderComponents.js

```diff
@@ -0,0 +1,210 @@
+  badgeText,
+  badgeTooltipText,
+}: {|
+  badgeText: string,
+  badgeTooltipText?: string,
+|}): Node {
+  return (
+    <Fragment>
+      {badgeText && !badgeTooltipText ? (
+        <Badge text={badgeText} type="info" position="top" />
+      ) : null}
+      {badgeText && badgeTooltipText ? (
+        <Tooltip accessibilityLabel="" text={badgeTooltipText} idealDirection="up">
+          <Badge text={badgeText} type="info" position="top" />
+        </Tooltip>
+      ) : null}
+    </Fragment>
+  );
+}
+
+export function PageHeaderHelperIconButton({
+  helperIconButton,
+}: {|
+  helperIconButton: {|
+    accessibilityLabel: string,
+    accessibilityControls: string,
+    accessibilityExpanded: boolean,
+    onClick: ({|
+      event:
+        | SyntheticMouseEvent<HTMLButtonElement>
+        | SyntheticKeyboardEvent<HTMLButtonElement>
+        | SyntheticMouseEvent<HTMLAnchorElement>
+        | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return (
+    <Box dangerouslySetInlineStyle={{ __style: { marginTop: '-1px' } }} rounding="circle">
+      <IconButton
+        accessibilityControls={helperIconButton.accessibilityControls}
+        accessibilityExpanded={helperIconButton.accessibilityExpanded}
+        accessibilityHaspopup
+        accessibilityLabel={helperIconButton.accessibilityLabel}
+        bgColor="lightGray"
+        icon="question-mark"
+        iconColor="darkGray"
+        onClick={helperIconButton.onClick}
+        size="xs"
+      />
+    </Box>
+  );
+}
+
+export function PageHeaderSubtext({
+  subtext,
+  helperLink,
+}: {|
+  subtext?: string,
+  helperLink?: {|
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return (
+    <Box display="none" smDisplay="block">
+      <Text inline>
+        {subtext}{' '}
+        {helperLink ? (
+          <Text inline underline>
+            <Link
+              accessibilityLabel={helperLink.accessibilityLabel}
+              href={helperLink.href}
+              onClick={helperLink.onClick}
+              target="blank"
+              inline
+            >
+              <span style={{ textDecoration: 'underline' }}>{helperLink.text}</span>
+            </Link>
+          </Text>
+        ) : null}
+      </Text>
+    </Box>
+  );
+}
+
+export function PageHeaderActionBlock({
+  primaryAction,
+  secondaryAction,
+  dropdownItems,
+  dropdownAccessibilityLabel = '',
+}: {|
+  primaryAction: ActionType,
+  secondaryAction?: ActionType,
+  dropdownItems?: $ReadOnlyArray<Node>,
+  dropdownAccessibilityLabel?: string,
+|}): Node {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef(null);
+
+  if (!dropdownItems || !dropdownAccessibilityLabel) {
```

Comment:
I think we can improve our guidance here…what if the user provides `primaryAction` and `secondaryAction` but only one dropdown item? What if they provide `primaryAction` and three dropdown items? Or what if the order of the dropdown items doesn't match `[primary, secondary]`? This API gives the user a lot of ways to shoot themselves in the foot, which isn't great. Consider how the dropdown items can be more tightly coupled to their respective actions — perhaps as part of the action props rather than a separate prop.

Reply:
DONE